crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1250,50 +1247,33 @@ mod tests {
     const MAC_EOL: &str = "\r";
     const UNIX_EOL: &str = "\n";
 
-    pub(crate) fn lex_source(source: &str) -> Vec<Tok> {
+    fn lex_source(source: &str) -> Vec<Tok> {
         let lexer = lex(source, Mode::Module);
-        lexer.map(|x| x.unwrap().0).collect()
+        lexer.map(|result| result.unwrap().0).collect()
     }
 
-    pub(crate) fn lex_jupyter_source(source: &str) -> Vec<Tok> {
+    fn lex_jupyter_source(source: &str) -> Vec<Tok> {
         let lexer = lex(source, Mode::Ipython);
         lexer.map(|x| x.unwrap().0).collect()
     }
 
-    #[test_case(UNIX_EOL)]
-    #[test_case(MAC_EOL)]
-    #[test_case(WINDOWS_EOL)]
-    fn test_ipython_escape_command_line_continuation_eol(eol: &str) {
+    #[test_case(UNIX_EOL, "unix")]
+    #[test_case(MAC_EOL, "mac")]
+    #[test_case(WINDOWS_EOL, "windows")]
+    fn test_ipython_escape_command_line_continuation_eol(eol: &str, platform: &str) {
+        let snapshot = format!("ipython_escape_command_line_continuation_{platform}_eol");
         let source = format!("%matplotlib \\{eol}  --inline");
-        let tokens = lex_jupyter_source(&source);
-        assert_eq!(
-            tokens,
-            vec![
-                Tok::IpyEscapeCommand {
-                    value: "matplotlib   --inline".to_string(),
-                    kind: IpyEscapeKind::Magic
-                },
-                Tok::Newline
-            ]
-        );
+        assert_debug_snapshot!(snapshot, lex_jupyter_source(&source));
     }
 
-    #[test_case(UNIX_EOL)]
-    #[test_case(MAC_EOL)]
-    #[test_case(WINDOWS_EOL)]
-    fn test_ipython_escape_command_line_continuation_with_eol_and_eof(eol: &str) {
+    #[test_case(UNIX_EOL, "unix")]
+    #[test_case(MAC_EOL, "mac")]
+    #[test_case(WINDOWS_EOL, "windows")]
+    fn test_ipython_escape_command_line_continuation_with_eol_and_eof(eol: &str, platform: &str) {
+        let snapshot =
+            format!("ipython_escape_command_line_continuation_with_{platform}_eol_and_eof",);
         let source = format!("%matplotlib \\{eol}");
```

Comment:
Nit: I prefer to avoid `test_case` and other macros except if necessary. I think I would have created a helper function here instead

```rust
fn iptyon_escape_command_line_continuation_with_eol(eol: &str) -> Vec<Token> {
	let source = format!("%matplotlib \\{eol}");
	lex_jupyter_source(&source)
}

#[test]
fn ipython_escape_command_line_continuation_with_ln() {
	assert_debug_snapshot(ipython_escape_command_line_continuation_with_eol("\n"))
}
```

It's slightly more code but requires no knowledge about any macros.